internal/statestore/backfill.go

```diff
@@ -140,6 +142,65 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+// AcknowledgeBackfill stores Backfill's last acknowledgement time.
+// Check on Backfill existence should be performed on Frontend side
+func (rb *redisBackend) AcknowledgeBackfill(ctx context.Context, id string) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return status.Errorf(codes.Unavailable, "AcknowledgeBackfill, id: %s, failed to connect to redis: %v", id, err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	currentTime := time.Now().UnixNano()
+	cmds := make([]interface{}, 0)
+	cmds = append(cmds, backfillLastAckTime, currentTime, id)
+
+	_, err = redisConn.Do("ZADD", cmds...)
```

Comment:
There's no need for the oddness with cmds, just do:
`_, err = redisConn.Do("ZADD", backfillLastAckTime, currentTime, id)`